src/category_theory/limits/shapes/bicones.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2017 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import category_theory.limits.cones
+import category_theory.structured_arrow
+import category_theory.fin_category
+
+/-!
+# Bicones
+
+Given a category `J`, a walking `bicone J` is a category whose objects are the objects of `J` and
+two extra vertices `bicone.left` and `bicone.right`. The morphisms are the morphisms of `J` and
+`left ⟶ j`, `right ⟶ j` for each `j : J` such that `⬝ ⟶ j` and `⬝ ⟶ k` commutes with each
+`f : j ⟶ k`.
+
+Given a diagram `F : J ⥤ C` and two `cone F`s, we can join them into a diagram `bicone J ⥤ C` via
+`bicone_mk`.
+
+This is used in `category_theory.flat_functors.preserves_finite_limits_of_flat`.
+-/
+
+universes v₁ u₁
+
+open category_theory.limits
+
+namespace category_theory
+section bicone
+variables (J : Type u₁)
+
+/-- Given a category `J`, construct a waling `bicone J` by adjoining two elements. -/
+@[derive decidable_eq, nolint has_inhabited_instance]
+inductive bicone (J : Type u₁)
+| left : bicone
+| right : bicone
+| diagram (val : J) : bicone
+
+instance fin_bicone [fintype J] [decidable_eq J] : fintype (bicone J) :=
+{ elems := [bicone.left, bicone.right].to_finset ∪ finset.image bicone.diagram (fintype.elems J),
+  complete := λ j, by { cases j; simp, exact fintype.complete j, }, }
+
+variables [category.{v₁} J] [∀ (j k : J), decidable_eq (j ⟶ k)]
+
+/-- The homs for a walking `bicone J`. -/
+@[nolint has_inhabited_instance]
```

Comment:
I personally don't love `nolint` for this linter, but I know others disagree. In this case though it's pretty easy to give examples (`bicone_hom.left_id` and `bicone.left` above) so perhaps these could be added?